tldw_Server_API/app/core/Evaluations/recipes/rag_retrieval_tuning_candidates.py

```
@@ -0,0 +1,328 @@
+    }
+    for key in retrieval_config:
+        normalized[key] = retrieval_config[key]
+
+    search_mode = str(normalized["search_mode"]).strip().lower()
+    if search_mode not in _ALLOWED_SEARCH_MODES:
+        raise ValueError(f"unsupported candidate knob: search_mode={search_mode}")
+    normalized["search_mode"] = search_mode
+    normalized["top_k"] = int(normalized["top_k"])
+    normalized["hybrid_alpha"] = float(normalized["hybrid_alpha"])
+    if not 0.0 <= normalized["hybrid_alpha"] <= 1.0:
+        raise ValueError("hybrid_alpha must be between 0.0 and 1.0.")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)